crates/bitwarden-core/src/uniffi_support.rs

```diff
@@ -0,0 +1,39 @@
+use uuid::Uuid;
+
+use crate::UniffiCustomTypeConverter;
+
+type DateTime = chrono::DateTime<chrono::Utc>;
+uniffi::custom_type!(DateTime, std::time::SystemTime);
+
+impl UniffiCustomTypeConverter for chrono::DateTime<chrono::Utc> {
+    type Builtin = std::time::SystemTime;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        Ok(Self::from(val))
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        obj.into()
+    }
+}
+
+uniffi::custom_type!(Uuid, String);
+
+impl UniffiCustomTypeConverter for Uuid {
+    type Builtin = String;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        Uuid::parse_str(val.as_str()).map_err(|e| e.into())
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        obj.to_string()
+    }
+}
+
+#[allow(dead_code)]
+#[derive(uniffi::Record)]
```

Comment:
Is this one of those workarounds to deal with types not being emitted by uniffi if they are not used in a record? If so I think we should add a comment mentioning why it's here